### PR TITLE
Test project summary

### PR DESCRIPTION
**Pull request by katieemelianova**
Not yet ready for merging
Rewrite some of the data collection
Add a test report to be included in the Quality Manual
To include changes in PR #36

### CODE REVIEW

File: project_report/__init__.py [hunk 1]
```diff
@@ -295,6 +301,13 @@ def get_pc_statistics(self):
                 pc_statistics['samples'].append(all_pc_statistics[3])
         return pc_statistics
 
+
+
+
+
+
+
+
```
Comment (mwhamgenomics) [inline project_report/__init__.py]: ?

File: project_report/__init__.py [hunk 2]
```diff
@@ -312,10 +325,19 @@ def yield_plot(self, sample_labels=False):
...
+
+
```
Comment (mwhamgenomics) [inline project_report/__init__.py]: ?

File: project_report/__init__.py [hunk 3]
```diff
@@ -44,41 +47,102 @@ def __init__(self, project_name):
...
+    def samples_for_project_lims(self):
+        if self._lims_samples_for_project is None:
+            self._lims_samples_for_project = self.lims.get_samples(projectname=self.project_name)
+        return self._lims_samples_for_project
 
-    def get_total_number_of_samples(self):
-        project = clarity.get_project(project_id=self.project_name)
-        return project.udf.get('Number of Quoted Samples')
+    @property
+    def samples_for_project_restapi(self):
```
Comment (tcezard) [inline project_report/__init__.py]: This should query the REST API only for delivered samples

File: project_report/__init__.py [hunk 4]
```diff
@@ -44,41 +47,102 @@ def __init__(self, project_name):
...
+        if self._database_samples_for_project is None:
+            self._database_samples_for_project = get_documents('aggregate/samples', match={"project_id": self.project_name}, paginate=False)
+            if not self._database_samples_for_project:
+                raise EGCGError('No samples found for project %s' % (self.project_name))
+        return self._database_samples_for_project
+
+    def get_sample(self, sample_name):
```
Comment (tcezard) [inline project_report/__init__.py]: Maybe rename this function to `get_lims_sample`

File: project_report/__init__.py [hunk 5]
```diff
@@ -44,41 +47,102 @@ def __init__(self, project_name):
...
 
-    def get_delivered_sample_names(self):
-        return [s.get('sample_id') for s in self.samples_from_lims]
+    def get_all_sample_names(self, modify_names=False):
+        if modify_names:
+            return [re.sub(r'[: ]', '_', s.name) for s in self.samples_for_project_lims]
+        return [s.name for s in self.samples_for_project_lims]
+
+    def get_samples_delivered(self):
+        sample_yields = [s.get('clean_yield_in_gb') for s in self.samples_for_project_restapi if s.get('clean_yield_in_gb')]
```
Comment (tcezard) [inline project_report/__init__.py]: This is not the definition of delivered samples.

File: project_report/__init__.py [hunk 6]
```diff
@@ -44,41 +47,102 @@ def __init__(self, project_name):
...
+    def get_species_found(self, sample):
+        sample_contamination = sample.get('species_contamination', {}).get('contaminant_unique_mapped')
+        if sample_contamination:
+            species_found = [s for s in sample_contamination if sample_contamination[s] > 500]
+            return species_found
+        return None
+
+    def get_library_workflow(self, samples):
+        library_workflow = set()
+        for sample in samples:
```
Comment (tcezard) [inline project_report/__init__.py]: By using the  sample names from the sample_delivered  this should be safe against samples with workflow not set.

File: project_report/__init__.py [hunk 7]
```diff
@@ -109,76 +173,299 @@ def read_metrics_csv(metrics_csv):
...
         return samples_to_info
 
+    def get_project_info(self):
+        sample_names = self.get_all_sample_names()
+        genome_versions = set()
+        species_submitted = set()
+        library_workflow = self.get_library_workflow(self.get_all_sample_names())
+        self.params['library_workflow'] = library_workflow
+
+        for sample in sample_names:
```
Comment (tcezard) [inline project_report/__init__.py]: This should also only use the sample delivered

File: project_report/templates/all_templates.html [hunk 8]
```diff
@@ -0,0 +1,143 @@
+{% macro yield_chart(params) -%}
+    <h3>Yield</h3>
+    <img id="chart1" class="charts" src="{{ params['yield_chart'] }}" />
+    <p class="justify_text">Yield is calculated as the yield in gigabases (Gb) after adapters have been
+    trimmed during processing. The yield Q30 is the quantity of reads
+     which have at least 75% of bases with a quality score of 30 or higher.</p>
```
Comment (tcezard) [inline project_report/templates/all_templates.html]: That is not the definition of Q30.

File: project_report/__init__.py [hunk 9]
```diff
@@ -121,6 +143,14 @@ def project_size_in_terabytes(self):
         project_size = self.get_folder_size(self.project_delivery)
         return (project_size/1000000000000.0)
 
+    def parse_date(self, date):
+        if not date:
+            return 'None'
+        d = parser.parse(date)
+        datelist = [d.year, d.month, d.day]
+        return '-'.join([str(i) for i in datelist])
```
Comment (mwhamgenomics) [inline project_report/__init__.py]: dateutil is not part of the standard library and is not currently maintained. See below for an implementation with datetime:
```python
>>> date = '2017-11-07T16:18:06.673000'
>>> d = datetime.strptime(date, '%Y-%m-%dT%H:%M:%S.%f')
>>> d.strftime('%Y-%m-%d')
'2017-11-07'
```

**Pull request merged**